src/simple_edg.rs

```diff
@@ -0,0 +1,158 @@
+/// the desired name of the vertex enum. This allows us to have multiple hardcoded EDGs in the
+/// same scope.
+/// ```
+/// simple_edg![
+///     [MyEDG1, MyVertex1]
+///     A => -> {B} .> {D};
+///     B => -> {D, C};
+///     C => ;
+///     D => -> {C};
+/// ];
+///
+/// edg_assert!([MyEDG1, MyVertex1] A, TRUE);
+/// edg_assert!([MyEDG1, MyVertex1] B, FALSE);
+/// ```
+/// The `edg_assert` macro allows the same naming functionality.
+macro_rules! simple_edg {
+    // Standard use
+    [ $( $v:ident => $( -> { $( $t:ident ),* } )* $( .> $n:ident )* );*; ] => {
+        simple_edg![[SimpleEDG, SimpleVertex] $( $v => $( -> { $( $t ),* } )* $( .> $n )* );*; ]
+    };
+    // Defines struct and enum
+    [ [ $edg_name:ident, $vertex_name:ident ] $( $v:ident => $( -> { $( $t:ident ),* } )* $( .> $n:ident )* );*; ] => {
+        #[derive(Hash, Clone, Eq, PartialEq, Debug)]
+        struct $edg_name;
+        #[derive(Hash, Clone, Eq, PartialEq, Debug)]
+        enum $vertex_name {
+            $( $v ),*
+        }
+        impl Display for $vertex_name {
+            fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+                write!(f, "{:?}", self)
+            }
+        }
+        impl Vertex for $vertex_name {}
+        impl ExtendedDependencyGraph<$vertex_name> for $edg_name {
+            fn succ(
+                &self,
+                vertex: &$vertex_name,
+            ) -> HashSet<Edges<$vertex_name>> {
+                simple_edg![@match vertex_name=$vertex_name, vertex $( $v => $( -> { $( $t ),* } )* $( .> $n )* );*;]
+            }
+        }
+    };
+    // Constructs match expression
+    [ @match vertex_name=$vertex_name:ident, $vertex:ident $( $v:ident => $( -> { $( $t:ident ),* } )* $( .> $n:ident )* );*; ] => {
+        match $vertex {
+            $($vertex_name::$v => {
+                #[allow(unused_mut)]
+                let mut successors = HashSet::new();
+                $(successors.insert(Edges::HYPER(HyperEdge {
+                    source: $vertex_name::$v,
+                    targets: vec![$($vertex_name::$t),*],
+                }));)*
+                $(successors.insert(Edges::NEGATION(NegationEdge {
+                    source: $vertex_name::$v,
+                    target: $vertex_name::$n,
+                }));)*
+                successors
+            }),*
+        }
+    };
+}
+
+/// Defines an assertion which test the assignment of a vertex in an EDG using the
+/// distributed certain zero algorithm.
+/// This macro is intended to be used in conjunction with the `simple_edg` macro.
+///
+/// # Example
+/// Simple usage:
+/// ```
+/// simple_edg![
+///     A => .> B;
+///     B => -> {};
+/// ];
+///
+/// edg_assert!(A, FALSE);
+/// edg_assert!(B, TRUE);
+/// ```
+/// Note that TRUE/FALSE must be capitalized.
+///
+/// # Worker count
+/// You can set the worker count by supplying a third argument to the marco. The default
+/// number of workers are 3.
```

Comment:
probably wise to not have it default to 1 👍

Reply:
I am not sure about that. Testing the multi-threadedness is also important for the `distributed_certain_zero` algorithm. For instance, we have three tests that does not work when worker count = 3, but do when worker count = 1. These tested have been marked with `#[ignore]` in this PR and should hopefully be fixed in #111. But I only found them because I changed our default worker count to 3 (from 1).

Reply:
Yes, and the discovery of these failing tests is exactly why I think it's good that we have changed the default to 3

Reply:
Oh, I think I misread your comment then :D